### PR TITLE
Add placeholder value for keras learning phase if required

### PR DESCRIPTION
**Pull request by rnett**
On imported tensorflow models that were originally Keras models, some layers (like GRUs) require a `keras_learning_phase` placeholder value, which is a boolean that is true if the model is training.  

This is known in `AbstractSession#output`, so if we are trying to get a `keras_learning_phase` placeholder we return that value.

The cast is a bit hacky, but I don't think it will break anything.

### Tests
Ran ND4J tests, no egregious failures, `AbstractSession` doesn't show up in any of the failures.

### CODE REVIEW

**Comment by AlexDBlack** [comment]
Also maybe we should make this configurable at some point too...
Like if user has some "learning rate placeholder" by some other name, they can set it's name in the training configuration, and have it auto-filled (if not otherwise provided).
